src/main/java/org/jabref/gui/collab/DatabaseChangeMonitor.java

```diff
@@ -62,10 +63,22 @@ public DatabaseChangeMonitor(BibDatabaseContext database,
             }
         });
 
+        LibraryTab.DatabaseNotification notificationPane = libraryTab.getNotificationPane();
         addListener(changes -> notificationPane.notify(
                 IconTheme.JabRefIcons.SAVE.getGraphicNode(),
                 Localization.lang("The library has been modified by another program."),
-                List.of(new Action(Localization.lang("Dismiss changes"), event -> notificationPane.hide()),
+                List.of(new Action(Localization.lang("Dismiss changes"), event -> libraryTab.getNotificationPane().hide()),
+                        new Action(Localization.lang("Accept changes"), event -> {
+                            // Perform all accepted changes
+                            NamedCompound ce = new NamedCompound(Localization.lang("Merged external changes"));
+                            for (DatabaseChangeViewModel change : changes) {
+                                if (change instanceof EntryChangeViewModel || change.isAccepted()) {
+                                    change.makeChange(database, ce);
+                                }
+                            }
+                            ce.end();
```

Comment:
Please add the following line:

```java
// TODO: panel.getUndoManager().addEdit(ce);
```

(Source org.jabref.gui.collab.ChangeDisplayDialog#ChangeDisplayDialog - line 94)

Otherwise, the NamedCompound is obsolete somehow.